### PR TITLE
[MLIR][Presburger] Implement PresburgerRelation::convertVarKind and a…

### PR DESCRIPTION
**Pull request by iambrj**
…dd unit test for IntegerRelation::convertVarKind

### CODE REVIEW

**Comment by iambrj** [comment]
I noticed while implementing this that the interface for `IntegerRelation::convertVarKind` is slightly different from `PresburgerSpace::convertVarKind` from @Groverkss's recent d222b69 commit. I feel it is worth making them consistent, let me know what you think @Superty @Groverkss.

File: mlir/lib/Analysis/Presburger/PresburgerRelation.cpp [hunk 1]
```diff
@@ -38,6 +38,21 @@ void PresburgerRelation::insertVarInPlace(VarKind kind, unsigned pos,
   space.insertVar(kind, pos, num);
 }
 
+void PresburgerRelation::convertVarKind(VarKind srcKind, unsigned srcPos,
+                                        unsigned num, VarKind dstKind,
+                                        unsigned dstPos) {
+  assert(srcKind != dstKind && "cannot convert variables to the same kind");
```
Comment (Groverkss) [inline mlir/lib/Analysis/Presburger/PresburgerRelation.cpp:44]: Please add an assert checking if either of srcKind/dstKind is a VarKind::Local.

**Review by Groverkss — APPROVED** [review]
LGTM. Just a minor nit.

**Comment by Groverkss** [comment]
> I noticed while implementing this that the interface for `IntegerRelation::convertVarKind` is slightly different from `PresburgerSpace::convertVarKind` from @Groverkss's recent [d222b69](https://github.com/llvm/llvm-project/commit/d222b69093d69e034b7f01d9ceedce28d48fbb0c) commit. I feel it is worth making them consistent, let me know what you think @Superty @Groverkss.

IntegerRelation::convertVarKind needs to be changed to be consistent with PresburgerSpace.

**Pull request merged**